streams/src/main/java/org/apache/kafka/streams/kstream/internals/foreignkeyjoin/BaseForeignKeyJoinProcessorSupplier.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.kstream.internals.foreignkeyjoin;
+
+import org.apache.kafka.streams.processor.ProcessorSupplier;
+
+abstract public class BaseForeignKeyJoinProcessorSupplier<K, V> implements ProcessorSupplier<K, V> {
```

Comment:
would remove this class and avoid protected final here. Doesn't seem like a convincing java programming style.

Reply:
I will change this - but I do appreciate any advice on how I should changee that. I'll post my updated code and we can determine if it's any better, or if there is a more specific approach I should follow.